### PR TITLE
Add skeleton Oak Containers Launcher for testing

### PR DESCRIPTION
**Pull request by jul-sh**
Ref #4023

### CODE REVIEW

File: oak_containers_launcher/src/server.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#[tonic::async_trait]
+impl Launcher for LauncherServerImplementation {
+    type GetOakSystemImageStream = GetImageResponseStream;
+    type GetContainerBundleStream = GetImageResponseStream;
+
+    async fn get_oak_system_image(
+        &self,
+        _request: Request<()>,
+    ) -> Result<Response<Self::GetOakSystemImageStream>, tonic::Status> {
+        let response_stream = async_stream::try_stream! {
```
Comment (andrisaar) [inline oak_containers_launcher/src/server.rs:48]: Interesting; I guess this yields one chunk that is empty as a response.

Side question: If you leave out the `yield`, would the stream immediately terminate with no data?

**Review by andrisaar — APPROVED** [review]
Thanks; I was just talking to Conrad about a launcher earlier, so this was perfect timing.

**Review by conradgrobler — APPROVED** [review]
Nice, thanks.

File: oak_containers_launcher/src/server.rs (hunk 1 above)
Comment (jul-sh) [reply oak_containers_launcher/src/server.rs:48]: It'd come out as a type error, the macro expects us to actually yield :s.

The macro isn't from tonic, so edges are a bit rough.

**Pull request merged**